File: src/main/java/ru/hasa/springbootapp/service/RoleServiceImpl.java
package ru.hasa.springbootapp.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.hasa.springbootapp.dao.RoleDao;
import ru.hasa.springbootapp.model.Role;

import java.util.List;

@Service
public class RoleServiceImpl implements RoleService{
    @Autowired
    private final RoleDao roleDao;

    public RoleServiceImpl(RoleDao roleDao) {
        this.roleDao = roleDao;
    }

    @Override
    public List<Role> getAllRoles() {
        return roleDao.getAllRoles();
    }
}

File: src/main/kotlin/ru/hasa/springbootapp/service/RoleServiceImpl.kt
package ru.hasa.springbootapp.service

import org.springframework.stereotype.Service
import ru.hasa.springbootapp.dao.RoleDao
import ru.hasa.springbootapp.model.Role

@Service
class RoleServiceImpl(private val roleDao: RoleDao) : RoleService {
    override fun getAllRoles(): List<Role> = roleDao.getAllRoles()
}
